experimental/FTheoryTools/src/TateModels/constructors.jl

```diff
@@ -239,6 +254,37 @@ function global_tate_model(auxiliary_base_ring::MPolyRing, auxiliary_base_gradin
 end
 
 
+function _construct_generic_sample(base_grading::Matrix{Int64}, base_vars::Vector{String}, d::Int, ais::Vector{T}) where {T<:MPolyRingElem}
```

Comment:
To avoid duplication of code and thus potential future errors upon edits, should we instead have a separate `_construct_toric_sample` and `_construct_generic_sample` function that each of HypersurfaceModels, TateModels, and WeierstrassModels calls upon in their own way?

Reply:
Similar comments apply here as to the version of this function in HypersurfaceModels. The code replacing the loop is probably less important, since so many of the entries are set by hand in this case.